backend/src/api/model/block/mutations.rs

```diff
@@ -0,0 +1,355 @@
+use futures::StreamExt;
+use pin_utils::pin_mut;
+use juniper::{GraphQLInputObject, GraphQLObject, Nullable};
+
+use crate::{api::{Context, Id, err::{ApiResult, invalid_input}}, dbargs};
+use crate::db::types::Key;
+use super::{BlockValue, VideoListLayout, VideoListOrder, super::realm::Realm};
+
+
+impl BlockValue {
+    pub(crate) async fn add_text(
+        realm: Id,
+        index: i32,
+        block: NewTextBlock,
+        context: &Context,
+    ) -> ApiResult<Realm> {
+        context.require_moderator()?;
+
+        let index = index as i16;
+
+        let realm = realm.key_for(Id::REALM_KIND)
+            .ok_or_else(|| invalid_input!("`realm` does not refer to a realm"))?;
+
+        Self::make_room_for_block(realm, index, context).await?;
+
+        context.db
+            .execute(
+                "insert into blocks (realm_id, index, type, title, text_content)
+                    values ($1, $2, 'text', $3, $4)",
+                &[&realm, &index, &block.title, &block.content],
+            )
+            .await?;
+
+        Realm::load_by_key(realm, context)
+            .await?
+            .ok_or_else(|| invalid_input!("`realm` does not refer to a valid realm"))
```

Comment:
I thought about a similar function; I don't remember the details, though. I decided against it, though, because there doesn't really seem to be a great abstraction here. Like, sometimes the check looks maybe slightly different because it's more convenient or sth. :shrug: 

Do you want me to add it or will you take it as is? :wink: 

Regarding the second point: I guess? :shrug: :sweat_smile: I need the key anyway, for the `make_room` function, and this way I can immediately return the realm. What's the advantage in your proposed change? :thinking:

Reply:
We can do the `try_load_by_key` or whatever in some of those famous "various improvements" PRs later :P 
And yeah nevermind the other stuff